library/network_connections.py

```diff
@@ -2340,6 +2341,36 @@ def check_name(self, idx, name=None):
             return None
         return f
 
+    def forget_nm_connection(self, path):
+        """
+        Forget a NetworkManager connection by loading the path of the deleted
+        profile. This inverts the effect of loading a profile with
+        `NM_CONTROLLED=no` earlier, which made NetworkManager ignore the
+        device.
+
+        This does ot use the Python libnm bindings because they might not be
+        present on the system, since the module is currently operating for the
```

Comment:
Would it make sense to install those bindings every time that there is a NetworkManager package present on the system, even if the module is not operating for the nm provider? What package is it actually, `python-gobject-base`?

Reply:
> Would it make sense to install those bindings every time that there is a NetworkManager package present on the system, even if the module is not operating for the nm provider? 

Installing it seems to be rather intrusive. This is now the same code I proposed for https://bugzilla.redhat.com/show_bug.cgi?id=1832897 therefore I do not see much harm in the current way. Another advantage of this approach is that it keeps the provider specific code separate.

> What package is it actually, `python-gobject-base`?
Yes or `python3-gobject-base`.